value_helper.go

```diff
@@ -424,6 +425,15 @@ func kuzuValueToGoValue(kuzuValue C.kuzu_value) (any, error) {
 		return kuzuStructValueToGoValue(kuzuValue)
 	case C.KUZU_MAP:
 		return kuzuMapValueToGoValue(kuzuValue)
+	case C.KUZU_DECIMAL:
+		var outString *C.char
+		status := C.kuzu_value_get_decimal_as_string(&kuzuValue, &outString)
+		if status != C.KuzuSuccess {
+			return nil, fmt.Errorf("failed to get string value of decimal type with status: %d", status)
+		}
+		goDecimal, _ := decimal.NewFromString(C.GoString(outString))
+		defer C.kuzu_destroy_string(outString)
```

Comment:
This does not require a deferred destroy. It can be destroyed immediately at this point.